src/main/java/sparksoniq/jsoniq/runtime/iterator/postfix/PredicateIterator.java

```diff
@@ -43,12 +43,16 @@
     private RuntimeIterator _iterator;
     private RuntimeIterator _filter;
     private Item _nextResult;
+    private long _position;
+    private boolean _mustMaintainPosition;
+    private DynamicContext _filterDynamicContext;
```

Comment:
We need to chain dynamic contexts. The context passed to the filter must be a new context, so as not to modify in place the parent context passed to the left hand side and potentially other places.